diff --git a/node_modules/@babel/runtime/helpers/objectDestructuringEmpty.js b/node_modules/@babel/runtime/helpers/objectDestructuringEmpty.js
index 5b405e0..c10d307 100644
--- a/node_modules/@babel/runtime/helpers/objectDestructuringEmpty.js
+++ b/node_modules/@babel/runtime/helpers/objectDestructuringEmpty.js
@@ -1,4 +1,4 @@
 function _objectDestructuringEmpty(obj) {
-  if (obj == null) throw new TypeError("Cannot destructure " + obj);
+  if (obj == null) console.log("Cannot destructure " + obj);
 }
 module.exports = _objectDestructuringEmpty, module.exports.__esModule = true, module.exports["default"] = module.exports;
\ No newline at end of file
