diff --git a/node_modules/@notifee/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java b/node_modules/@notifee/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java
index 40bea64..8019a2b 100644
--- a/node_modules/@notifee/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java
+++ b/node_modules/@notifee/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java
@@ -39,9 +39,9 @@ public class NotifeeApiModule extends ReactContextBaseJavaModule implements Perm
   // This method was removed upstream in react-native 0.74+, replaced with invalidate
   // We will leave this stub here for older react-native versions compatibility
   // ...but it will just delegate to the new invalidate method
-  public void onCatalystInstanceDestroy() {
-    invalidate();
-  }
+  // public void onCatalystInstanceDestroy() {
+  //   invalidate();
+  // }
 
   // This method was added in react-native 0.74 as a replacement for onCatalystInstanceDestroy
   // It should be marked @Override but that would cause problems in apps using older react-native
